core/server/models/base/listeners.js

```diff
@@ -120,3 +120,10 @@ common.events.on('settings.active_timezone.edited', function (settingModel, opti
             });
     });
 });
+
+/**
+ * WHEN locale changes, we reload theme translations
+ */
+common.events.on('settings.default_locale.edited', function () {
```

Comment:
This should live in the i18n service. The listener here is only responsible for data/model changes.

Reply:
Event listener to reload theme translations for locale change moved to i18n service.